From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 2 Aug 2026 15:53:49 +1000
Subject: [PATCH] dt-bindings: clock: fsl,imx8m-anatop: Allow the syscon
 compatible

The i.MX8MQ anatop block holds the PLL monitor and SCCG divider
registers that consumers outside the clock driver need to reach, so it
has to be usable as a syscon. It was described that way until commit
f98c2dfedb73 ("arm64: dts: imx8m: align anatop with bindings") dropped
the compatible to match this schema.

Allow "fsl,imx8mq-anatop", "syscon" so those consumers work again.

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
---
 Documentation/devicetree/bindings/clock/fsl,imx8m-anatop.yaml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Documentation/devicetree/bindings/clock/fsl,imx8m-anatop.yaml b/Documentation/devicetree/bindings/clock/fsl,imx8m-anatop.yaml
index bbd22e9..925e33e 100644
--- a/Documentation/devicetree/bindings/clock/fsl,imx8m-anatop.yaml
+++ b/Documentation/devicetree/bindings/clock/fsl,imx8m-anatop.yaml
@@ -23,6 +23,9 @@ properties:
               - fsl,imx8mn-anatop
               - fsl,imx8mp-anatop
           - const: fsl,imx8mm-anatop
+      - items:
+          - const: fsl,imx8mq-anatop
+          - const: syscon
 
   reg:
     maxItems: 1
